activerecord/lib/active_record/statement_cache.rb

```diff
@@ -49,8 +49,8 @@ def initialize values
 
       def sql_for(binds, connection)
         val = @values.dup
-        binds = connection.prepare_binds_for_database(binds)
-        @indexes.each { |i| val[i] = connection.quote(binds.shift) }
+        quoted_binds = connection.quoted_binds(binds)
```

Comment:
As I mentioned, I don't want to rely on any new methods outside of the connection.